File: Two Pointers/LC_11_Container_With_Most_Water.py
# Written by: Jian-Yuan Lin
# Date: 2022.11.17

class Solution:
    def maxArea(self, height: List[int]) -> int:
        l, r = 0, len(height) - 1
        ans = 0
        while l < r:
            minH = min(height[l], height[r])
            area = minH * (r - l)
            ans = max(ans, area)
            if height[l] < height[r]:
                l += 1
            else:
                r -= 1
        return ans

File: Two Pointers/LC_11_Container_With_Most_Water.cpp
// Written by: Jian-Yuan Lin
// Date: 2022.11.17

class Solution {
public:
    int maxArea(vector<int>& height) {
        int ans = 0;
        int l = 0, r = height.size() - 1;

        while (l < r){
            int h = min(height[l], height[r]);
            ans = max(ans, h * (r-l));
            if (height[l] < height[r]){
                l++;
            }
            else{
                r--;
            }
        }
        return ans;
    }
};